lib/ipam/ipam_block.go

```diff
@@ -21,56 +21,34 @@ import (
 	"net"
 	"reflect"
 
+	"github.com/projectcalico/libcalico-go/lib/apis/v3"
 	log "github.com/sirupsen/logrus"
 
 	"github.com/projectcalico/libcalico-go/lib/backend/model"
 	cnet "github.com/projectcalico/libcalico-go/lib/net"
 )
 
-const (
-	blockSize = 64
-)
-
-type ipVersion struct {
-	Number            int
-	TotalBits         int
-	BlockPrefixLength int
-	BlockPrefixMask   net.IPMask
-}
-
-var ipv4 ipVersion = ipVersion{
-	Number:            4,
-	TotalBits:         32,
-	BlockPrefixLength: 26,
-	BlockPrefixMask:   net.CIDRMask(26, 32),
-}
-
-var ipv6 ipVersion = ipVersion{
-	Number:            6,
-	TotalBits:         128,
-	BlockPrefixLength: 122,
-	BlockPrefixMask:   net.CIDRMask(122, 128),
-}
-
 // Wrap the backend AllocationBlock struct so that we can
 // attach methods to it.
 type allocationBlock struct {
 	*model.AllocationBlock
+	numAddresses int
```

Comment:
Not a big deal, but couldn't we do this without storing this information in a new field?

e.g. just call `numAddresses(..)` on the block cidr whenever we need this information or expose a function that can calculate this on the fly? Doesn't seem too computationally complex.